libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -87,6 +89,8 @@ pub(crate) struct BTCReceiveSwap {
     persister: Arc<crate::persist::db::SqliteStorage>,
     chain_service: Arc<dyn ChainService>,
     payment_receiver: Arc<dyn Receiver>,
+    current_tip: Mutex<u32>,
```

Comment:
Since we already have the persister, it might be simpler to use
```rust
self.persister.node_info()?.block_height
```

like in https://github.com/breez/breez-sdk/pull/812/files#diff-dfcf1c8c08fb243da9a4e02f631600d2e5a5c827fbbed56c8e98daadd4347483R253

Reply:
Good point but when tested I have noticed the greenlight block sometimes lag behind the mempool space block.
Almost always there is a 1 block lag. Also when the node is not connected for a while it takes some time to catchup and gl shows the tip it is aware of which lags a lot behind for a while.
Perhaps if gl will expose a listener to block change it will make more sense as even if lags a bit behind it will be consistent. What do you think?

Reply:
I wasn't aware the GL tip was lagging. Yes, then it makes sense the way you did it.

I'll abandon my other PR using `self.persister.node_info()?.block_height` as it would suffer from the same problem.